### PR TITLE
OCPBUGS-15843: Finish FIPS + ISO support

### PR DESCRIPTION
**Pull request by cgwalters**
fips: use --rbind for sysroot

This is necessary for the Live ISO case where we have a `/sysroot/etc`
mount that we need to pick up.

---

initramfs/fips: Pretend we're RHEV-H for verifying Live ISO

A while ago, the RHEV-H folks added special case code
to the dracut initramfs FIPS path, because how the bootloader(s)
pass `BOOT_IMAGE` is less reliable in a Live (particularly PXE)
path.

Pretend we're RHEV-H here by creating a symlink from the place
they picked in `/run` into what I consider the canonical location
of `/usr/lib/modules/$kver`.

And we should definitely change dracut to look there in the
future as it would drop out a lot of special case hacks in that code.

Requires: https://github.com/coreos/coreos-assembler/pull/3599
xref https://issues.redhat.com/browse/OCPBUGS-15843

---

### CODE REVIEW

**Comment by cgwalters** [comment]
Leaving draft for now as this is necessary for https://issues.redhat.com/browse/OCPBUGS-15843 but not sufficient.

**Comment by cgwalters** [comment]
OK now tested in concert with https://github.com/coreos/coreos-assembler/pull/3599

File: overlay.d/05rhcos/usr/lib/dracut/modules.d/40rhcos-fips/rhcos-fips.sh [hunk 1]
```diff
@@ -69,6 +69,17 @@ finish() {
...
+    # module will find the kernel binary.  TODO change dracut to look in /usr/lib/modules/$(uname -r)
+    # directly.
+    if test -f /etc/coreos-live-initramfs; then
+        # See the dracut source
+        rhevh_livedir=/run/initramfs/live
+        mkdir -p "${rhevh_livedir}"
+        # Why "vmlinuz0"?  I have no idea; it's what the dracut fips module uses.
+        ln -sr /usr/lib/modules/$(uname -r)/vmlinuz ${rhevh_livedir}/vmlinuz0
+    fi
+
```
Comment (zaneb) [inline overlay.d/05rhcos/usr/lib/dracut/modules.d/40rhcos-fips/rhcos-fips.sh:82]: Since for testing purposes there is a workaround for this bit, I'd be inclined to defer this to another PR and try to get the --rbind thing merged.

Comment (cgwalters) [reply overlay.d/05rhcos/usr/lib/dracut/modules.d/40rhcos-fips/rhcos-fips.sh:82]: With just the first commit, all that happens in FIPS mode is the system dies in boot in a different way...doesn't seem useful?

Comment (cgwalters) [reply overlay.d/05rhcos/usr/lib/dracut/modules.d/40rhcos-fips/rhcos-fips.sh:82]: > Since for testing purposes there is a workaround for this bit

Oh, are you thinking of the `rd.fips.skipkernel` kernel argument?  Yes true...

**Comment by cgwalters** [comment]
Lifting draft since this can land independently of https://github.com/coreos/coreos-assembler/pull/3599

**Review by travier — APPROVED** [review]
Used https://askubuntu.com/questions/1122975/difference-between-rbind-and-bind-in-mounting to figure out the difference here.

/lgtm
